MapboxNavigation/TopBannerViewController.swift

```diff
@@ -0,0 +1,433 @@
+        let bottom = informationStackBottomPin
+        //bottom is taken care of as part of steps TVC show/hide
+        
+        NSLayoutConstraint.activate([top, leading, trailing, bottom])
+    }
+    
+    private func setupInformationStackView() {
+        addInstructionsBanner()
+        informationStackView.addArrangedSubviews(secondaryChildren)
+        for child in informationChildren {
+            child.leadingAnchor.constraint(equalTo: informationStackView.leadingAnchor).isActive = true
+            child.trailingAnchor.constraint(equalTo: informationStackView.trailingAnchor).isActive = true
+        }
+    }
+
+
+    public func displayStepsTable() {
+        dismissStepsTable()
+        
+        guard let progress = routeProgress, let parent = parent else {
+            return
+        }
+        
+        
+        let controller = StepsViewController(routeProgress: progress)
+        controller.delegate = self
+        
+        delegate?.topBanner?(self, willDisplayStepsController: controller)
+        embed(controller, in: stepsContainer) { (parent, child) -> [NSLayoutConstraint] in
+            child.view.translatesAutoresizingMaskIntoConstraints = false
+            let pinningConstraints = child.view.constraintsForPinning(to: self.stepsContainer)
+            let hideConstraints = self.stepsContainerHide
+            
+            return pinningConstraints + hideConstraints + self.stepsContainerConstraints
+        }
+        stepsViewController = controller
+        isDisplayingSteps = true
+        
+        parent.view.layoutIfNeeded()
+        
+        
+        let stepsInAnimation = {
+            NSLayoutConstraint.deactivate(self.stepsContainerHide)
+            NSLayoutConstraint.activate(self.stepsContainerShow)
+            
+            
+            let finally: (Bool) -> Void = { _ in
+                self.delegate?.topBanner?(self, didDisplayStepsController: controller)
+            }
+            
+            UIView.animate(withDuration: 0.35, delay: 0.0, options: [.curveEaseOut], animations: parent.view.layoutIfNeeded, completion: finally)
+        }
+
+        hideSecondaryChildren(completion: stepsInAnimation)
+        
+    }
+    
+    public func dismissStepsTable(completion: CompletionHandler? = nil) {
+        guard let parent = parent, let steps = stepsViewController  else { return }
+        parent.view.layoutIfNeeded()
+        
+        delegate?.topBanner?(self, willDismissStepsController: steps)
+
+
+        NSLayoutConstraint.deactivate(stepsContainerShow)
+        NSLayoutConstraint.activate(stepsContainerHide)
+        
+        let complete = {
+            self.delegate?.topBanner?(self, didDismissStepsController: steps)
+            completion?()
+        }
+        
+        UIView.animate(withDuration: 0.35, delay: 0.0, options: [.curveEaseInOut], animations: parent.view.layoutIfNeeded, completion: { _ in
+            if !self.isDisplayingPreviewInstructions {
```

Comment:
There's a leak triggering every time the steps table view controller gets dismissed. I assume the culprit is around here. Inline the completion closure for readability and weakify self since these are strong refs.

Reply:
Okay, I think I nailed it down. Unable to confirm with instrumentation, as the leaks that remain I'm also detecting on `master`, but every vulnerable surface area has been weakified.